app/resources/views.py

```diff
@@ -45,18 +46,76 @@ def add():
                             user_id=int(current_user.get_id()))
         db.session.add(resource)
         db.session.commit()
-        return redirect(url_for('resources.index'))
+        return redirect(url_for('resources.show', resource_id=resource.id))
     return render_template('resources/add_resource.html', form=form)
 
 
 @resources.route('/resource/<int:resource_id>')
 @login_required
-def show_resource(resource_id):
-    """
-    Show the resource with the given id, the id is an integer.
-    """
+def show(resource_id):
     resource = Resource.query.get_or_404(resource_id)
     address = Address.query.get(resource.address_id)
     user = User.query.get(resource.user_id)
+    current_user_id = int(current_user.get_id())
+    # TODO: base template for reviews.
     return render_template('resources/view_resource.html', resource=resource,
-                           address=address, user=user)
+                           address=address, user=user,
+                           current_user_id=current_user_id)
+
+
+@resources.route('/resource/<int:resource_id>/writeareview',
```

Comment:
also, let's not start this route with `/resource/`if it's already in the resource blueprint